lightning/src/chain/chainmonitor.rs

```diff
@@ -387,7 +598,12 @@ where C::Target: chain::Filter,
 				monitor.load_outputs_to_watch(chain_source);
 			}
 		}
-		entry.insert(MonitorHolder { monitor });
+		entry.insert(MonitorHolder {
+			monitor,
+			pending_monitor_updates: Mutex::new(pending_monitor_updates),
+			channel_perm_failed: AtomicBool::new(false),
+			last_chain_persist_height: AtomicUsize::new(0),
```

Comment:
I think if persistence returned `Ok(())`, we can set this to the current height

Reply:
Note that `last_chain_persist_height` is only relevant if a `ChainSync` update is pending, which it can't be until we hit `process_chain_data`, at which point we'll init it.

Reply:
I think we can init it to the current height in any case - we only care about it in the context of chainsync persistences, which this (obviously) isn't one of.